### ISSUE TITLE
CatBoostError: catboost/libs/train_lib/dir_helper.cpp:20: Can't create train working dir:

### ISSUE THREAD

**Issue by jd-jagmohandhiman** [catboost/catboost#1891]
Problem: I am running catboost on Databricks cluster. Databricks Production cluster is very secure and we cannot create new directory on the go as a user. But we can have pre-created directories. I am passing below parameter for my CatBoostClassifier.
_`CatBoostClassifier(train_dir='dbfs/FileStore/files/')`_
It does not work and throws below error.
`CatBoostError: catboost/libs/train_lib/dir_helper.cpp:20: Can't create train working dir`
catboost version:0.24.2
Operating System: Databricks cluster (Unix)

**Comment by ek-ak** [comment]
You are right, it is not a good solution. Now there is no ways to fit catboost with snapshots but creating new folders.  Catboost creates 'tmp', 'learn', 'test', ('test1', ..., 'testN' if you have more than 1 test) folders in train_dir if they are not exist.
You can pre-create this directories and all should work.

**Comment by zahraegh** [comment]
is there any solution to this problem? I am using databricks which does not allow any writing to DBFS due to security issues. I tried to give it an mlflow path, but still, get the same error. Is there a way to be able to specify mlflow path for this `train_dir`?

**Comment by andrey-khropov** [comment]
> is there any solution to this problem? I am using databricks which does not allow any writing to DBFS due to security issues. I tried to give it an mlflow path, but still, get the same error. Is there a way to be able to specify mlflow path for this `train_dir`?

Do you run CatBoost for Spark package or some other CatBoost package?

**Comment by zahraegh** [comment]
> Do you run CatBoost for Spark package or some other CatBoost package?
I just pip install catboost in databricks

**Comment by sergylog** [comment]
Hey! Any news on which path to choose?

**Comment by harshitapal06** [comment]
Hi,
The train_dir will create in the present working directory.

**Comment by andrey-khropov** [comment]
> Hi @ek-ak
> 
> Catboost creates a folder known as "catboost_info". Is there any method through which CATBOOST does not creates this folder.
> 
> I don't want to have this folder while training the model. 
> 
> I have tried allow_wrting_file=False but it doesn't work

@HiiHarshita 

It is very strange, this option has been for a long time and seems to work without problems for me. I've rechecked with the last release (1.2.2) and it still works. Can you give a detailed example how to reproduce this issue?

**Comment by andrey-khropov** [comment]
> Hey! Any news on which path for `train_dir` to choose?

@sergylog 

Is this a DataBricks specific question? You can read about working with files in DataBricks [here](https://docs.databricks.com/en/files/index.html).

CatBoost python package uses standard POSIX file API (on macOS or Linux) so use paths without schema.
